disnake/ext/commands/params.py

```diff
@@ -57,12 +65,91 @@
 
     AnySlashCommand = Union[InvokableSlashCommand, SubCommand]
 
+    from typing_extensions import TypeGuard
+
+if sys.version_info >= (3, 9):
+    from typing import Annotated
+else:
+    Annotated = object()
+
 T = TypeVar("T", bound=Any)
+TypeT = TypeVar("TypeT", bound=Type[Any])
+CallableT = TypeVar("CallableT", bound=Callable[..., Any])
 ChoiceValue = Union[str, int, float]
 Choices = Union[List[OptionChoice], List[ChoiceValue], Dict[str, ChoiceValue]]
 TChoice = TypeVar("TChoice", bound=ChoiceValue)
 
-__all__ = ("ParamInfo", "Param", "param", "option_enum", "Current")
+__all__ = (
+    "ParamInfo",
+    "Param",
+    "param",
+    "inject",
+    "option_enum",
+    "register_injection",
+    "converter_method",
+)
+
+
+def issubclass_(obj: Any, tp: Union[TypeT, Tuple[TypeT, ...]]) -> TypeGuard[TypeT]:
+    if not isinstance(obj, type) or not isinstance(tp, (type, tuple)):
+        return False
+    return issubclass(obj, tp)
+
+
+def evaluate_forwardref(annotation: Any, globalns: Dict[str, Any]) -> Any:
+    if isinstance(annotation, (str, ForwardRef)):
+        if isinstance(annotation, str):
+            annotation = ForwardRef(annotation)
+
+        try:
+            annotation = annotation._evaluate(globalns, globalns, recursive_guard=frozenset())  # type: ignore
+        except Exception as e:
+            annotation = annotation.__forward_arg__
+            warnings.warn(f"Cannot parse annotation: {annotation!r}")
```

Comment:
In case someone is using `TYPE_CHECKING`. I just did it because I wanted the error to show up later and possibly ignore any annotations which won't be used but now that I think about it it's probably better to raise errors here.